protocol/nats_jetstream/v2/receiver.go

```diff
@@ -163,6 +169,28 @@ func (c *Consumer) applyOptions(opts ...ConsumerOption) error {
 	return nil
 }
 
+// pullSubscriptionProcessor allows the pull subscriber to control some aspects of the fetch calls
+func (c *Consumer) pullSubscriptionProcessor(ctx context.Context, natsSub *nats.Subscription, ps *PullSubscriber) {
+	for {
+		// Wait until external or internal context done
+		select {
+		case <-ctx.Done():
+			return
+		case <-c.internalClose:
+			return
+		default:
+			msgs, err := ps.FetchCallback(natsSub)
+			if err != nil || msgs == nil {
+				continue
+			}
```

Comment:
Not sure I'm a fan of just dropping the error message here, unless it is logged internall before it is called?

Also there's no need for the nil check on `msgs` - `range` over a nil slice is a valid operation and just skips.

Reply:
returning error to calling function.